### DISCUSSION TITLE
Create a Global Configurations Options stored in MySQL table config or similar [done]

### DISCUSSION THREAD

**Discussion by magicbug** [magicbug/Cloudlog#630] [General]
Its become apparent going forward Cloudlog will need Global Configuration options that go out across the whole installation., this could be things like QRZ logins, CSS Stylesheet (whether you have light or dark, default date formats etc etc.

This would only be assigned to "Admins" which will become more useful when I finally get round to making cloudlog better for multi-user, coming soon with the creation of Logbooks #609 to sit in front of Station Profiles, will also help us move away from static configuration files!

**Comment by m0pcb** [comment]
This is of particular value to users of hosted instances who cannot rely on
editing config files. Presenting these options to a UI page will also be
an important feature. :)

73, Iain AD5XI / M0PCB

**Comment by magicbug** [comment]
This feature is now coming together I've added a migration file that creates an options table similar to what WordPress uses and an option for yes/no to auto load the option.

Next up is coding to automatically load the options and start using them around Cloudlog.

**Comment by magicbug** [comment]
OK just committed a change which has the following

- A new library called **OptionsLib** this can be accessed anywhere in cloudlog now with `$this->optionslib->`
- The library is set to automatically load within Cloudlog and loads all options set to autoload "yes" within the database any option is stored within the config system and can be accessed using `$this->config->item('option_<option_value>')`
- I've created a function in the **optionslib** that gets the correct theme choice and then it loads it at the moment the value in the options for theme is "default" but if you change it you'll see it change for even logged out users, each logged in user can have a different theme regardless of the default.
- You can call `$this->optionslib->get_option('option_name');` passing the option name and it will return the options value, use this for accessing non autoload items, you don't need to call the models ever.

**Todos**

- Need some code for saving new options
- Need some frontend to build out the "Options" area in Cloudlog (Have some ideas on this will be tomorrows job)

I'll get these added to the developer notes once I've got the other bits done.

@AndreasK79 @dg9vh @daccle @jsb2092

**Comment by AndreasK79** [comment]
Great work Peter. This will be much better for all. Now I just need to study the code a bit to see how it can be used for other settings.
